internal/monitors/ecs/ecs.go

```diff
@@ -0,0 +1,262 @@
+	shouldIgnore map[string]bool
+	imageFilter filter.StringFilter
+}
+
+// Configure the monitor and kick off volume metric syncing
+func (m *Monitor) Configure(conf *Config) error {
+	var err error
+	m.imageFilter, err = filter.NewBasicStringFilter(conf.ExcludedImages)
+	if err != nil {
+		return errors.Wrapf(err, "Could not load excluded image filter")
+	}
+
+	m.conf = conf
+	m.timeout = time.Duration(conf.TimeoutSeconds) * time.Second
+	m.client = &http.Client{
+		Timeout: m.timeout,
+	}
+	m.ctx, m.cancel = context.WithCancel(context.Background())
+
+	isRegistered := false
+
+	utils.RunOnInterval(m.ctx, func() {
+		if !isRegistered {
+			m.containers = map[string]ecs.Container{}
+			m.shouldIgnore = map[string]bool{}
+			m.fetchContainers()
+			
+			isRegistered = true
+		}
+
+		m.fetchStatsForAll()
+
+	}, time.Duration(conf.IntervalSeconds)*time.Second)
+
+	return nil
+}
+
+// Fetch all containers in the task and load it to the monitor
+func (m *Monitor) fetchContainers() {
+	body, err := getMetadata(m.client, m.conf.MetadataEndpoint)
+	if err != nil {
+		logger.WithError(err).Error("Failed to read ECS container data - %v", err)
+		return
+	}
+
+	task := ecs.TaskMetadata{}
+	if err := json.Unmarshal(body, &task); err != nil {
+		logger.WithFields(log.Fields{
+			"error":    err,
+		}).Error("Could not parse ecs metadata json")
+		return
+	}
+	m.taskDimensions = task.GetDimensions()
+
+	for i := range task.Containers {
+		if (m.imageFilter == nil ||
+			!m.imageFilter.Matches(task.Containers[i].Image)) &&
+			 // Containers that are specified in the task definition are of type NORMAL. This will filter out all AWS internal containers
+			task.Containers[i].Type == "NORMAL" {
+			m.containers[task.Containers[i].DockerId] = task.Containers[i]
+			m.shouldIgnore[task.Containers[i].DockerId] = false
+		} else {
+			m.shouldIgnore[task.Containers[i].DockerId] = true
+		}
+	}
+}
+
+// Fetch a container with given container docker ID and load it to the monitor
+// If the container is successfully received, return true. Else, return false
+func (m *Monitor) fetchContainer(dockerId string) bool {
+	body, err := getMetadata(m.client, fmt.Sprintf("%s/%s", m.conf.MetadataEndpoint, dockerId))
+
+	if err != nil {
+		logger.WithError(err).Error("Failed to read ECS container data - %v", err)
+		return false
+	}
+
+	var container ecs.Container
+
+	if err := json.Unmarshal(body, &container); err != nil {
+		logger.WithFields(log.Fields{
+			"error":    err,
+		}).Error("Could not parse ecs container json")
+		return false
+	}
+
+	if (m.imageFilter != nil && m.imageFilter.Matches(container.Image)) &&
+		container.Type == "NORMAL" {
+		return false
+	}
+
+	m.containers[dockerId] = container
+	return true
+}
+
+func (m *Monitor) fetchStatsForAll() {
+	_, cancel := context.WithTimeout(m.ctx, m.timeout)
+	
+	body, err := getMetadata(m.client, m.conf.StatsEndpoint)
+
+	if err != nil {
+		cancel()
+		logger.WithError(err).Error("Failed to read ECS stats - %v", err)
```

Comment:
You don't need to format the error into the error string if you use `WithError(err)`.